libs/wingc/src/jsify.rs

```diff
@@ -1825,27 +1794,33 @@ impl<'ast> Visit<'ast> for PreflightTypeRefVisitor<'ast> {
 		self.env = backup_env;
 	}
 
+	fn visit_type_annotation(&mut self, node: &'ast crate::ast::TypeAnnotation) {
+		if let TypeAnnotationKind::UserDefined(u) = &node.kind {
+			let root = u.root.name.as_str();
+			let mut arr = vec![root];
+			arr.extend(u.fields.iter().map(|f| f.name.as_str()));
+
+			let path = arr.join(".");
+			self.add_ref(path);
+		}
+
+		visit::visit_type_annotation(self, node);
+	}
+
 	fn visit_reference(&mut self, node: &'ast Reference) {
```

Comment:
I think the visitor's trait's default implementation of `visit_reference` calls `visit_type_annotation` in case of a static method (in case of a `TypeMember` reference). So it should work.

Reply:
Yes but it also calls `visit_type_annotation` in many cases I don't want to account for.